modules/drivers/bigquery-cloud-sdk/src/metabase/driver/bigquery_cloud_sdk/query_processor.clj

```diff
@@ -431,6 +431,26 @@
     [_ _ expr]
     (with-temporal-type (hsql/call bigquery-fn expr) :timestamp)))
 
+(defmethod sql.qp/->honeysql [:bigquery-cloud-sdk :convert-timezone]
+  [driver [_ arg target-timezone source-timezone]]
+  (let [timestamp    (partial hsql/call :timestamp)
+        datetime     (partial hsql/call :datetime)
+        clause       (sql.qp/->honeysql driver arg)
+        timestamptz? (hx/is-of-type? clause "timestamp")]
+    (when (and timestamptz? source-timezone)
+      (throw (ex-info (tru "`timestamp` columns shouldn''t have a `source timezone`")
+                      {:type            qp.error-type/invalid-query
+                       :target-timezone target-timezone
+                       :source-timezone source-timezone})))
+    (let [source-timezone (or source-timezone (qp.timezone/results-timezone-id))]
+      (cond-> clause
+        (not timestamptz?)
+        (timestamp source-timezone)
+        true
+        (datetime target-timezone)
+        true
+        (with-temporal-type :datetime)))))
```

Comment:
Given the two `true`s I would refactor this into an `->` expression, combined with an `if` like you did for other drivers